client/tsconfig.json

```diff
@@ -9,6 +9,7 @@
     "noImplicitAny": false,
     "noEmit": true,
     "noFallthroughCasesInSwitch": true,
+    // "noUncheckedIndexedAccess": true,
```

Comment:
?

Reply:
Oops: I'll remove this and add an issue - this was a typescript quirk I noticed while developing: by default typescript assumes array index access will always return an object of the correct type, when it can also return `undefined` - there were some cases where I was wondering why typescript wasn't complaining about my obviously incorrect code. We should turn this on generally, but there's a few other places that need to be fixed first.